lms/djangoapps/discussion_api/api.py

```diff
@@ -123,3 +127,78 @@ def get_thread_list(request, course_key, page, page_size):
 
     results = [ThreadSerializer(thread, context=context).data for thread in threads]
     return get_paginated_data(request, results, page, num_pages)
+
+
+def get_comment_list(request, thread_id, endorsed, page, page_size):
+    """
+    Return the list of comments in the given thread.
+
+    Parameters:
+
+    request: The django request object used for build_absolute_uri and
+      determining the requesting user.
+    thread_id: The id of the thread to get comments for.
+    endorsed: Boolean indicating whether to get endorsed or non-endorsed
+      comments (or None for all comments). Must be None for a discussion thread
+      and non-None for a question thread.
+    page: The page number (1-indexed) to retrieve
+    page_size: The number of threads to retrieve per page
+
+    Returns:
+
+    A paginated result containing a list of comments; see
+    discussion_api.views.CommentViewSet for more detail.
+    """
+    response_skip = page_size * (page - 1)
+    try:
+        cc_thread = Thread(id=thread_id).retrieve(
+            recursive=True,
+            user_id=request.user.id,
+            mark_as_read=True,
+            response_skip=response_skip,
+            response_limit=page_size
+        )
+    except CommentClientRequestError:
+        raise Http404
+
+    course_key = CourseLocator.from_string(cc_thread["course_id"])
+    course = _get_course_or_404(course_key, request.user)
```

Comment:
This check cannot happen before the CS request, because the course_id comes from the thread data.